### PR TITLE
CloudSQLite: add support for httptimeout being configurable

### PR DESCRIPTION
**Pull request by nick4598**
The current default in SQLite code is 600 seconds. 

This will improve performance in cases where we do timeout when receiving blocks. There are some examples in the v2 checkpoint agent, which show upwards of 10 minutes wasted on a single request. These requests will get retried more quickly if we hit our timeout faster.

I opted for 60 seconds as the default, but lower may be even more beneficial especially in write cases. As a faster http timeout could mean a client releases the write lock faster.

### CODE REVIEW

**Comment by nick4598** [comment]
> what was the previous default?

It is 600 seconds in SQLite code I believe.

**Comment by kabentley** [comment]
> It is 600 seconds in SQLite code I believe.

Do the Azure guidelines for retries give a suggested value? Even 60 seconds seems long.

**Comment by kabentley** [comment]
Here's what the Azure documentation says about blob storage:
<html>
<body>


ServerTimeout | None | Server timeout interval for the request (value is rounded to seconds). If not specified, it will use the default value for all requests to the server. Usually, the best option is to omit this setting so that the server default is used.
-- | -- | --



</body>
</html>

i don't know what that means, but if we're always going to retry, it seems like waiting 60 seconds is a waste (what's the probability that it will succeed after 60 seconds)?

**Comment by nick4598** [comment]
> Here's what the Azure documentation says about blob storage:
> 
> ServerTimeout	None	Server timeout interval for the request (value is rounded to seconds). If not specified, it will use the default value for all requests to the server. Usually, the best option is to omit this setting so that the server default is used.
> i don't know what that means, but if we're always going to retry, it seems like waiting 60 seconds is a waste (what's the probability that it will succeed after 60 seconds)?

Yeah I also had trouble finding some useful information regarding http timeouts..

For what its worth, I do see in SEQ a log entry which succeeded after 30 seconds:
 17:42:10.957 | r1_104 [] [30381ms] (http=201) (rc=0) HTTP/1.1 201 Created
 Its probably pretty likely that we'd finish that request faster than 30 seconds even if we made it timeout and then retry.
 
 I also found this: https://www.ibm.com/docs/en/sva/7.0.0?topic=cpc-timeout-settings-http-https-communication
 WebSEAL: "is a high performance, multi-threaded Web server that applies fine-grained security policy to resources in the Security Access Manager protected Web object space." They use 60 seconds for intra-connection timeout.. which is sort of what we're doing here with the http timeout.
 
 I also worry about making this too short and then maybe a particularly large manifest combined with poor internet connection means they're timing out even though work is actually being done. At the moment, cachesize httpTimeout and nRequests isn't configurable for consumers of itwinjs-core ( I think). Should settings like that be added to IModelHost?

**Comment by kabentley** [comment]
i don't think it's worth trying to add this to IModelHost, because nobody will have any clue what to set it to. I can see setting the default to 60 seconds and experimenting with it in the Checkpoint agent.

File: iModelCore/BeSQLite/PublicAPI/BeSQLite/CloudSqlite.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ struct CloudCache {
     Utf8String m_name;
     std::vector<CloudContainer*> m_containers;
 
-    BE_SQLITE_EXPORT CloudResult InitCache(Utf8StringCR name, Utf8StringCR rootDir, int64_t cacheSize = 0, int64_t nRequest = 0, bool verboseCurlLog = false);
+    BE_SQLITE_EXPORT CloudResult InitCache(Utf8StringCR name, Utf8StringCR rootDir, int64_t cacheSize = 0, int64_t nRequest = 0, int64_t httpTimeout = 0, bool verboseCurlLog = false);
```
[the remark this answers is not in the document]
Comment (nick4598) [reply iModelCore/BeSQLite/PublicAPI/BeSQLite/CloudSqlite.h]: changed it

**Review by kabentley — APPROVED** [review]

**Pull request merged**